pkg/controller/servicebindingrequest/reconciler.go

```diff
@@ -166,6 +167,10 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 		logger.Info("Searching DeploymentConfig objects matching labels")
 
 		deploymentConfigListObj := &osappsv1.DeploymentConfigList{}
+		err = resourcepoll.WaitUntilResourcesFound(r.client, &searchByLabelsOpts, deploymentConfigListObj)
```

Comment:
I assume that the reason this was put in place is because it indeed can take some time for the deployment config objects to appear, but now every time this gets executed it'll wait at least the first interval for each worker, which seems to block the whole process for each reconcile attempt.

A somehow easier approach would be a delayed re-enqueue at this point, pretty much moving the waiting process from inside the process to the queue itself (as in re-enqueue the ServiceBindingRequest object in 5 seconds if related objects are not present). 

Yet another approach would be listen for changes in DeploymentConfig at the operator level, and enqueue the related ServiceBindingRequest once a relevant object comes from the stream. This is more work, but also means that all the events of related objects are processed, and work on ServiceBindingRequest reconciliation would be re-attempted automatically.

Reply:
+1 for the delayed re-enqueue approach